app/javascript/components/miq-ae-customization/index.jsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState, useEffect } from 'react';
+import { FormSpy } from '@data-driven-forms/react-form-renderer';
+import { Button } from 'carbon-components-react';
+import MiqFormRenderer, { useFormApi } from '@@ddf';
+import PropTypes from 'prop-types';
+import createSchema from './miq-ae-customization-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const MiqAeCustomization = ({ dialogRecord, dialogTypes }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [initialValues, setInitialValues] = useState(undefined);
+
+  useEffect(() => {
+    if (dialogRecord && dialogRecord.id) {
+      http.get(`/miq_ae_customization/old_dialogs_edit_get/${dialogRecord.id}/`).then((recordValues) => {
+        if (recordValues) {
+          setInitialValues(recordValues);
+        }
+      });
+    } else {
+      const temp = {};
+
+      if (dialogRecord.name) {
+        if (dialogRecord.name) {
+          temp.name = dialogRecord.name;
+        }
+        if (dialogRecord.description) {
+          temp.description = dialogRecord.description;
+        }
+        if (dialogRecord.content) {
+          temp.content = dialogRecord.content;
+        }
+        if (dialogRecord.dialog_type) {
+          temp.dialog_type = dialogRecord.dialog_type;
+        }
+      }
+
+      if (!dialogRecord.content) {
+        temp.content = '---\n';
+      }
+      setInitialValues(temp);
+    }
+    setIsLoading(false);
+  }, [dialogRecord]);
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+
+    const actionType = dialogRecord.id ? 'edit' : 'create';
+
+    const data = {
+      action: actionType,
+      name: values.name,
+      description: values.description,
+      dialog_type: values.dialog_type,
+      content_data: values.content,
+      old_data: initialValues,
+    };
+
+    data.button = dialogRecord.id ? 'save' : 'add';
+    let request;
+    if (dialogRecord && dialogRecord.id) {
+      request = http.post(`/miq_ae_customization/old_dialogs_update_react/${dialogRecord.id}`, data);
+    } else {
+      request = http.post(`/miq_ae_customization/old_dialogs_update_react/`, data);
+    }
+
+    request
+      .then(() => {
+        const confirmation = dialogRecord.id ? 'Dialog Edited' : 'Dialog Created';
+        const message = sprintf(
+          confirmation,
+          values.name
+        );
+        miqRedirectBack(message, 'success', '/miq_ae_customization/explorer');
+      })
+      .catch(miqSparkleOff);
+  };
```

Comment:
also reduced few lines in here - 
```
const onSubmit = (values) => {
    miqSparkleOn();
    const isEdit = !!(dialogRecord && dialogRecord.id);

    const params = {
      action: isEdit ? 'edit' : 'create',
      name: values.name,
      description: values.description,
      dialog_type: values.dialog_type,
      content_data: values.content,
      old_data: data.initialValues,
      button: dialogRecord.id ? 'save' : 'add',
    };

    const request = isEdit
      ? http.post(`/miq_ae_customization/old_dialogs_update_react/${dialogRecord.id}`, params)
      : http.post(`/miq_ae_customization/old_dialogs_update_react/`, params);

    request
      .then(() => {
        const confirmation = isEdit ? 'Dialog Edited' : 'Dialog Created';
        miqRedirectBack(sprintf(confirmation, values.name), 'success', '/miq_ae_customization/explorer');
      })
      .catch(miqSparkleOff);
  };
```

Reply:
maybe use this as a global variable since its being reused in many functions - 
`const isEdit = !!(dialogRecord && dialogRecord.id);`